src/Microsoft.Health.Dicom.Core/Features/Query/QueryService.cs

```diff
@@ -63,6 +76,32 @@ public class QueryService : IQueryService
             return new QueryResourceResponse(responseMetadata);
         }
 
+        private HashSet<CustomTagFilterDetails> RetrieveSuupportedCustomTagsForQueryResourceType(IReadOnlyList<CustomTagStoreEntry> customTags, QueryResource queryResource)
+        {
+            HashSet<CustomTagFilterDetails> ret = new HashSet<CustomTagFilterDetails>();
+
+            foreach (CustomTagStoreEntry customTag in customTags)
+            {
+                DicomTag[] result;
+                DicomTag dicomTag;
+                if (customTag.Status.Equals(CustomTagStatus.Added) && _dicomTagPathParser.TryParse(customTag.Path, out result))
```

Comment:
We need to move this method CustomTagService. It needs to expose GetActiveTagsAtLevel(QueryResourceLevel)